snappy/security.go

```diff
@@ -20,94 +20,241 @@
 package snappy
 
 import (
-	"encoding/json"
+	"bufio"
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"regexp"
 	"strings"
 
-	"gopkg.in/yaml.v2"
-
 	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/helpers"
 	"github.com/ubuntu-core/snappy/logger"
 	"github.com/ubuntu-core/snappy/pkg"
+	"github.com/ubuntu-core/snappy/policy"
+	"github.com/ubuntu-core/snappy/release"
 )
 
-type apparmorJSONTemplate struct {
-	Template      string   `json:"template"`
-	PolicyGroups  []string `json:"policy_groups"`
-	PolicyVendor  string   `json:"policy_vendor"`
-	PolicyVersion float64  `json:"policy_version"`
+type errPolicyNotFound struct {
+	// type of policy, e.g. template or cap
+	PolType string
+	// apparmor or seccomp
+	PolKind *securityPolicyType
+	// name of the policy
+	PolName string
 }
 
-type securitySeccompOverride struct {
-	Template      string   `yaml:"security-template,omitempty"`
-	PolicyGroups  []string `yaml:"caps,omitempty"`
-	Syscalls      []string `yaml:"syscalls,omitempty"`
-	PolicyVendor  string   `yaml:"policy-vendor"`
-	PolicyVersion float64  `yaml:"policy-version"`
+func (e *errPolicyNotFound) Error() string {
+	return fmt.Sprintf("could not find specified %s: %s (%s)", e.PolType, e.PolName, e.PolKind)
```

Comment:
Dropped the original comment here, as the message feels fine after re-reading it in context.